firstvoices/backend/tests/test_apis/test_games_api.py

```diff
@@ -0,0 +1,138 @@
+import json
+
+import pytest
+from django.core.cache import caches
+from rest_framework.test import APIClient
+
+from backend.models.constants import Role, Visibility
+from backend.tests.factories import (
+    CharacterFactory,
+    DictionaryEntryFactory,
+    get_site_with_member,
+)
+from backend.tests.test_apis.base_api_test import BaseSiteContentApiTest
+from backend.views.games_views import CACHE_KEY_WORDSY
+
+
+class TestWordsyEndpoint(BaseSiteContentApiTest):
+    """
+    Tests for wordsy endpoint
+    """
+
+    API_LIST_VIEW = "api:wordsy-list"
+
+    def setup_method(self):
+        self.client = APIClient()
+        self.site, self.member_user = get_site_with_member(
+            Visibility.PUBLIC, Role.LANGUAGE_ADMIN
+        )
+        self.client.force_authenticate(user=self.member_user)
+        CharacterFactory.create(site=self.site, title="a")
+        CharacterFactory.create(site=self.site, title="b")
+        CharacterFactory.create(site=self.site, title="c")
+
+    @pytest.mark.django_db
+    def test_list_empty(self):
+        # Overriding since there is no pagination class for this view
+        response = self.client.get(self.get_list_endpoint(site_slug=self.site.slug))
+
+        assert response.status_code == 200
+        response_data = json.loads(response.content)
+
+        assert response_data["orthography"] == ["a", "b", "c"]
+        assert response_data["words"] == []
+        assert response_data["validGuesses"] == []
+        assert response_data["solution"] == ""
+
+    @pytest.mark.django_db
+    def test_all_invalid_words(self):
```

Comment:
just double checking, if there are not enough words to play the wordsy game (e.g. too few possibles words + solutions), is this handled on the FE? or does it need to be handled on the BE?

Reply:
Yes, the enough words validation is handled on the FE side.